test/dshell/depsprot.d

```diff
@@ -0,0 +1,19 @@
+import dshell;
+void main()
+{
+    Vars.set("deps_file", "$OUTPUT_BASE.deps");
+    run("$DMD -m$MODEL -deps=$deps_file -Idshell/imports -o- $EXTRA_FILES/$TEST_NAME.d");
+    Vars.deps_file
+        .grep("^$TEST_NAME.*${TEST_NAME}_default")
+        .grep("private")
+        .enforceMatches("Default import protection in dependency file should be 'private'");
+    Vars.deps_file
+        .grep("^$TEST_NAME.*${TEST_NAME}_public")
+        .grep("public")
+        .enforceMatches("Public import protection in dependency file should be 'public'");
+    Vars.deps_file
+        .grep("^$TEST_NAME.*${TEST_NAME}_private")
+        .grep("private")
+        .enforceMatches("Private import protection in dependency file should be 'private'");
+    rm(Vars.deps_file);
```

Comment:
I know this is ported from the old tests, but there's actually no big reason to clean-up these temporary files. They are in the results directory anyhow and cleaning up individually takes time and sometimes even fails on Windows, so moving forward I would just not care about these temporary files, but eventually run dmd in a temporary directory which can then be purged. What do you think?

Reply:
Well I think I fixed the issue on windows: https://github.com/dlang/dmd/pull/9838

Not removing a test's output files would increase the disk-space requirement to run the entire test suite without cleaning in between.  Although, this may not be a concern depending on the size of the test artifacts.  If the test artifacts only take of 50% of the space of all the other files required to run the tests, then probably not a big concern.

The other concern is the added logic/places for error.  I think this can be fixed by instead putting all the output files in a sub-directory.  Then the logic to cleanup the tests could be moved into a common location that just cleans the whole subdirectory, which we could leave as an option if the user is concerned with disk space.  What about that?

Reply:
```bash
#Clean dmd repo:

[marler8997@nixos:~/ddev1/dmd]$ du -hs .
151M	.

#After making dmd:

[marler8997@nixos:~/ddev1/dmd]$ du -hs .
190M	.

#After running the full test suite:

[marler8997@nixos:~/ddev1/dmd]$ du -hs .
224M	.

```

Reply:
Ok, I've removed the `rm` calls in the tests themselves, and left in a comment in `d_do_test.d` that indicates how we would remove the files in between tests if we wanted:
```D
rmdirRecurse(testOutDir);
```